### PR TITLE
[RNGP] chore: migrate HeaderUtilTest to AssertJ

### PR DESCRIPTION
**Pull request by gustavoabel**
## Summary:

Issue: #45596 

## Changelog:

Migrated to AssertJ within file:
- ```HeaderUtilTest.kt```

## Test Plan:

Run ```./gradlew -p packages/gradle-plugin test```

### CODE REVIEW

**Comment by gustavoabel** [comment]
@cortinico, this is done!

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/HeaderUtilTest.kt [hunk 1]
```diff
@@ -24,26 +23,26 @@ class HeaderUtilTest {
...
   @Test
   fun nameStripDeletesTabs() {
-    assertEquals(TABULATION_STRIP_EXPECTED, HeaderUtil.stripHeaderName(TABULATION_TEST))
+    assertThat(HeaderUtil.stripHeaderName(TABULATION_TEST)).isEqualTo(TABULATION_STRIP_EXPECTED);
   }
 
   @Test
   fun nameStripRemovesExtraSymbols() {
-    assertEquals(BANNED_TEST_EXPECTED, HeaderUtil.stripHeaderName(NAME_BANNED_SYMBOLS_TEST))
+    assertThat(HeaderUtil.stripHeaderName(VALUE_BANNED_SYMBOLS_TEST)).isEqualTo(VALUE_BANNED_SYMBOLS_TEST);
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/HeaderUtilTest.kt:46]: This is wrong and is causing tests to fail:

```

> Task :packages:react-native:ReactAndroid:testDebugUnitTest

HeaderUtilTest > nameStripRemovesExtraSymbols FAILED
    org.junit.ComparisonFailure at HeaderUtilTest.kt:46
```

**Review by cortinico — CHANGES_REQUESTED** [review]

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/HeaderUtilTest.kt (hunk 1 above)
Comment (gustavoabel) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/HeaderUtilTest.kt]: @cortinico, adjusted!

**Comment by gustavoabel** [comment]
@cortinico, hey! i requested review again

**Review by cortinico — APPROVED** [review]

**Pull request closed without merging**